src/openexr.imageio/exrinput_c.cpp

```diff
@@ -1922,124 +1943,99 @@ OpenEXRCoreInput::read_native_deep_tiles(int subimage, int miplevel, int xbegin,
     ud.firstisfullread = nxtiles == 1 && nytiles == 1;
     ud.samplesset      = false;
 
-    exr_chunk_info_t cinfo;
-    exr_decode_pipeline_t decoder = EXR_DECODE_PIPELINE_INITIALIZER;
-    DecoderDestroyer dd(m_exr_context, &decoder);
-    // Note: the decoder will be destroyed by dd exiting scope
-    bool first   = true;
-    int curytile = firstytile;
+    std::atomic<bool> ok(true);
 
     // it is very slow to update the number of samples one at a time, so when not
     // only reading one line, make a pass and load all the sample counts
     if (!ud.firstisfullread) {
-        std::vector<uint32_t> all_samples;
-        all_samples.resize(npixels);
-
-        uint32_t* allsampdata = all_samples.data();
-        for (int ty = 0; ty < nytiles; ++ty, ++curytile) {
-            int curxtile = firstxtile;
-
-            int nlines = tileh;
-            if ((size_t(ty) * size_t(tileh) + size_t(tileh)) > height)
-                nlines = height - (ty * tileh);
-
-            for (int tx = 0; tx < nxtiles; ++tx, ++curxtile) {
-                rv = exr_read_tile_chunk_info(m_exr_context, subimage, curxtile,
-                                              curytile, miplevel, miplevel,
-                                              &cinfo);
-                if (rv != EXR_ERR_SUCCESS)
-                    break;
-
-                if (first) {
+        std::vector<uint32_t> all_samples(npixels);
+        parallel_for_2D(
```

Comment:
not sure if parallel_for_2D already specializes the case of pulling one tile or not, but the extra thread dispatch that is in ilmthread was definitely noticeable in the testing I had done...

Reply:
As a matter of fact, yes, if there is only one task in the loop, it's just performed by the calling thread.